### ISSUE TITLE
Possible to Delete CodePush Release?

### ISSUE THREAD

**Issue by DovahBrownies** [Microsoft/code-push#571]
Yes, I know this may come off as a very trivial question, but I can't seem to find any documentation explaining how to delete a CodePush release, especially from the AppCenter.

Any help?

**Comment by NickToropov** [comment]
Hi @DovahBrownies

according to https://docs.microsoft.com/en-us/appcenter/distribution/codepush/cli#a-namerolling-back-updates-rolling-back-updates it’s not possible. 
You should use 
`appcenter codepush rollback -a <owner-name>/<app-name> Staging` if you want to rollback broken release or 
`appcenter codepush patch -a <owner-name>/<app-name> --disabled` to disable it.
Also you can disable release from AppCenter UI by toggling `Enabled` option:
![screen shot 2018-05-18 at 5 36 02 pm](https://user-images.githubusercontent.com/31315772/40240867-4144b9dc-5ac2-11e8-92e8-e3deac8886d7.png)

**Comment by DovahBrownies** [comment]
I'm not sure why they'd leave out a feature like this. Perhaps to encourage devs to rollback updates to ensure users who already downloaded the flawed update aren't left stranded; Which I understand, but still a rollback and deleting should be readily available. Thank you.

**Comment by putuyoga** [comment]
It would be great if we can remove old code push releases

**Comment by yuri-kulikov** [comment]
Hi @cdunkel and @putuyoga!
There's a way to delete all the releases from a deployment by [this command](https://github.com/Microsoft/code-push/tree/master/cli#clearing-release-history):
```Bash
code-push deployment clear <appName> <deploymentName>
```
But the command is **irreversible**, so you should use it carefully.

**Comment by cdunkel** [comment]
Thanks @yuri-kulikov. This could be useful at least for cleaning up our testing deployment from time to time. Not sure I'd ever want to use it on production.

**Comment by thakichowdhury** [comment]
For those coming to this thread more recently, the command is now
```bash
appcenter codepush deployment clear -a <ownerName>/<appName> <deploymentName>
```

and the link to it in the docs is:
https://docs.microsoft.com/en-us/appcenter/distribution/codepush/cli#clearing-release-history

**Comment by leotm** [comment]
Just a question, will this ever be possible via https://appcenter.ms/users/username/apps/app-name/distribute/code-push - the screenshot @NickToropov mentions? https://github.com/Microsoft/code-push/issues/571#issuecomment-390228179

**Comment by jdnichollsc** [comment]
hi @yuri-kulikov 
exist a command to disable all history instead?

**Comment by haresh4d** [comment]
> For those coming to this thread more recently, the command is now
> 
> ```shell
> appcenter codepush deployment clear -a <ownerName>/<appName> <deploymentName>
> ```
> 
> and the link to it in the docs is:
> https://docs.microsoft.com/en-us/appcenter/distribution/codepush/cli#clearing-release-history

This command gives the Error :

'codepush deployment clear -a hmrathod/DemoApp Staging' failed with exception "Cannot read property 'body' of undefined"

**Comment by tverilytt** [comment]
> Thanks @yuri-kulikov. This could be useful at least for cleaning up our testing deployment from time to time. Not sure I'd ever want to use it on production.

I agree. It would be very helpful to be able to delete specific releases. And also being able to name the releases. As of now, they automatically get the name v1, v2,...?

Cheers
-jo

**Comment by A-Tokyo** [comment]
Any idea how to disable certain releases?

**Comment by bcanozter** [comment]
> Any idea how to disable certain releases?

` appcenter codepush patch -a <owner>/<appName> -l <label> -x `

this will disable the release with a given label

**Comment by A-Tokyo** [comment]
Thanks! @bcanozter , realized it after hitting a wall with `promote` for a while 😅
